From 44537e132c6df16e453034035b715efc806e2f11 Mon Sep 17 00:00:00 2001
From: Fazzi <faaris.ansari@proton.me>
Date: Thu, 27 Nov 2025 12:01:43 +0000
Subject: [PATCH] window: disable graphics offload

---
 src/app/window/mod.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/app/window/mod.rs b/src/app/window/mod.rs
index d4339fb..d25d1cf 100644
--- a/src/app/window/mod.rs
+++ b/src/app/window/mod.rs
@@ -31,7 +31,7 @@ impl Window {
     pub fn set_overlay(&self, widget: &impl IsA<Widget>) {
         let window = self.imp();
 
-        window.overlay.add_overlay(&graphics_offload(widget));
+        window.overlay.add_overlay(widget);
     }
 
     pub fn set_fullscreen(&self, fullscreen: bool) {
-- 
2.51.2

